src/lib/utils.py

```diff
@@ -512,7 +512,10 @@ def read_json_manifest(manifest_path: Path):
 def read_yaml_manifest(manifest_path: Path):
     """Read a YAML manifest from 'manifest_path'."""
     with manifest_path.open("r") as f:
-        return _yaml.load(f)
+        has_yaml_header = (_ := f.readline().strip()) == "---"
```

Comment:
You can put a comment on the document separator line:

```yaml
--- # what a nice document
foo: bar
```

Annoyingly `ruamel.yaml` can't quite round-trip this:

```python
>>> yaml = YAML()
>>> yaml.explicit_start = True
>>> yaml.dump(yaml.load("""--- # what a nice document\nfoo: bar"""), sys.stdout)
---
    # what a nice document
foo: bar
```

But given that I think you could relax this check a little:


```suggestion
        has_yaml_header = (_ := f.read(3)) == "---"
```

Reply:
Makes sense.